frontend/src/features/compatibility/RbiProxyForwardingPage.tsx

```diff
@@ -0,0 +1,124 @@
+import { useEffect } from 'react'
+import { useSearchParams } from 'react-router-dom'
+import { Flex, Stack, Text } from '@chakra-ui/react'
+import { useFeatureValue } from '@growthbook/growthbook-react'
+
+import { LANDING_ROUTE } from '~constants/routes'
+import Spinner from '~components/Spinner'
+
+import { useIsRbiIpCheck } from '~features/login/queries'
+
+const ALLOWED_FORWARDING_ROUTES = [
+  '/sgid/login',
+  '/mi/login',
+  '/api/v3/singpass/login',
+  '/api/v3/corppass/login',
+  '/api/v3/auth/sgid/login/callback',
+  '/api/v3/auth/sso/login/callback',
+]
+
+export const FORWARDING_DEFAULT_TIMEOUT = 1000 // Default timeout for forwarding requests from the user's browser
+export const FORWARDING_RBI_TIMEOUT = 10000 // Default timeout for forwarding requests from
+
+/**
+ * RbiProxyForwardingPage is a component that facilitates the handoff between
+ * a Remote Browser Isolation (RBI) proxy and the user's actual browser.
+ *
+ * This page helps the browser to select the most direct route to the application
+ * by introducing a configurable timeout that prioritizes redirecting the user's
+ * actual browser and delaying the RBI proxy's response (if applicable).
+ *
+ * - If the request is coming directly from the user's browser, a shorter delay is used.
+ * - If the request is coming through an RBI proxy a longer delay is used in the timeout.
+ * - If the redirect path is not explicitly allowed, it redirects to the landing page.
+ * @constructor
+ */
+
+export const RbiProxyForwardingPage = (): JSX.Element => {
+  const [params] = useSearchParams()
+
+  // Extract the forwarded request and the loop-preventing forwarded flag
+  const route = params.get('route')
+  const forwarded = params.get('forwarded')
+
+  const forwardingDefaultTimeout = useFeatureValue(
+    'forwarding-default-timeout',
+    FORWARDING_DEFAULT_TIMEOUT,
+  )
+  let forwardingRbiTimeout = useFeatureValue(
+    'forwarding-rbi-timeout',
+    FORWARDING_RBI_TIMEOUT,
+  )
+
+  // If the feature flag SDK is not working correctly, force a safe fallback value
+  if (
+    !forwardingRbiTimeout ||
+    forwardingRbiTimeout < forwardingDefaultTimeout
+  ) {
+    // Ensure that the RBI timeout is always longer than the default timeout
+    forwardingRbiTimeout = Math.max(
+      forwardingDefaultTimeout + FORWARDING_DEFAULT_TIMEOUT,
+      FORWARDING_RBI_TIMEOUT,
+    )
+  }
+
+  // Check if the request is coming from a known Remote Browser Isolation (RBI) proxy IP
+  const { data: isOnProxy } = useIsRbiIpCheck()
+
+  useEffect(() => {
+    // Set the redirect URL to the landing page by default
+    let redirectUrl = LANDING_ROUTE
+
+    // If the route is not explicitly allowed, redirect to the landing page
+    const allowedRoute = ALLOWED_FORWARDING_ROUTES.find((allowedRoute) =>
+      route?.startsWith(allowedRoute),
+    )
+
+    if (forwarded && route && allowedRoute) {
```

Comment:
Wondering if we could somehow monitor the the success of this redirect and whether there are people reaching this page unintentionally.

Could be a monitor on dd.

Reply:
good point. will add an error to the un-allowed code path to facilitate monitoring.

It would be a client-side error and can only be correlated on DD if the logs reach DD.

Reply:
Fixed in the [latest changeset](https://github.com/opengovsg/FormSG/pull/8581/commits/0d1affa72186ffe9d36a2887b24aeaed2c7f9517)!